qiskit_nature/operators/second_quantization/star_algebra.py

```diff
@@ -0,0 +1,86 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Sum Operator base interface."""
+
+from abc import ABC, abstractmethod
+
+from .multiply import MultiplyMixin
+from .tolerances import TolerancesMixin
+
+
+class StarAlgebraMixin(MultiplyMixin, TolerancesMixin, ABC):
```

Comment:
I want to use quantum_info's mixin as possible. Common mixin is better. Please see [`LinearOp`](https://github.com/Qiskit/qiskit-terra/blob/ac15bd1870eee159e65e28deb26cb05f88c50e10/qiskit/quantum_info/operators/linear_op.py) for example

Reply:
Okay I see. And the reason for currently duplicating these interfaces here is that https://github.com/Qiskit/qiskit-terra/pull/5617 is not merged yet, right?
Once that is part of Terra, we can replace the imports of our Mixin's with the `quantum_info`-ones?

Reply:
Yes. That is the plan.

Reply:
Alright I also just saw your other comment to Steve's question below. Therefore there are two aspects to this:
1. once the mixin's are merged in Terra we can use those for `Multiply` etc.
2. you will PR the `StarAlgebra` one into Terra because that is currently not supported there.

Since you mentioned `LinearMixin`: why don't you use that here, too?

Reply:
In `LinearMixin`, there are an parameter `qargs` which our operators don't support. I think we can modify our operator to use it.

Reply:
I see. `qargs` is used to specify which subset is being acted on, right? While the name does not fit our needs at all, do you see an example where this could be useful? I can't see a use in our addition but maybe in our multiplication if we want to do something like the following:
```python
op = FermionicOp('I' * 4)
plus = FermionicOp('+', qargs=0)
new_op = op @ plus
# new_op.label should now be '+III'
```
Is this what this could look alike?

Reply:
Yeah. I agree. `qargs` might be useful. However, we don't have enough time in this PR. I may enhance FermionicOp after this PR.

Reply:
Sounds good to me :+1: I'll resolve this for now then.